supla-client/Tools/build_ios.sh

```diff
@@ -16,78 +16,124 @@
+  exit 1
 fi
 
-BUILD_DIR="../"$BUILD_CFG
+SDK_VERSION=${SDK}
+BUILD_DIR="${SCRIPT_DIR}/../${BUILD_CFG}"
 DEVELOPER=`xcode-select -print-path`
+IOS_OPENSSL_LIB_INC="$OPENSSL"
 
 if [ ! -d $BUILD_DIR ]; then
-    echo "Build dir does not exists! " $BUILD_DIR;
-    exit;
+    echo "[ERROR] Build dir does not exists! " $BUILD_DIR
+    exit 1
 fi
 
+echo "[INFO] Supla client will be prepared for sdk version '${SDK}' inside ${BUILD_CFG} config"
+
 build()
 {
-	ARCH=$1
-        PARAMS=""
-
-        if [[ "${ARCH}" == "armv7" || "${ARCH}" == "armv7s" ]]; then
-           TARGET_CPU="ARM"
-        else
-           TARGET_CPU=`echo -n ${ARCH} | awk '{ print toupper($0) }'`
-        fi
+  ARCH=$1
+  PLATFORM=$2
+  PARAMS=""
+  TARGET_DIR="${PLATFORM}_${ARCH}"
+  LOG_FILE="${TARGET_DIR}/build.log"
   
-	if [[ "${ARCH}" == "i386" || "${ARCH}" == "x86_64" ]]; then
-		PLATFORM="iPhoneSimulator"
-                PARAMS="-DTARGET_OS_SIMULATOR=1 -DTARGET_CPU_${TARGET_CPU} "
-	else
-		PLATFORM="iPhoneOS"
-                PARAMS="-DTARGET_OS_EMBEDDED=1 -DTARGET_CPU_${TARGET_CPU} "
-	fi
-
-        cd $BUILD_DIR
-
-        export CROSS_TOP="${DEVELOPER}/Platforms/${PLATFORM}.platform/Developer"
-        export CROSS_SDK="${PLATFORM}${SDK_VERSION}.sdk"
-
-        export PARAMS=${PARAMS}"-DSRPC_WITHOUT_IN_QUEUE -DSRPC_WITHOUT_OUT_QUEUE -DSPROTO_WITHOUT_OUT_BUFFER -DUSE_DEPRECATED_EMEV_V1 -DTARGET_OS_IOS=1 -fembed-bitcode -I${IOS_OPENSSL_LIB_INC} -arch ${ARCH} -isysroot ${CROSS_TOP}/SDKs/${CROSS_SDK} -miphoneos-version-min=${SDK_VERSION}"
-        
-        make clean
-        make
-
-        mkdir -p ${ARCH}
-        mv libsupla-client.a ${ARCH}
-       
-};
-
-CWD=`pwd`
+  echo "[INFO] Building '${PLATFORM}' for architecture '${ARCH}'"
 
-build "armv7"
-build "armv7s"
-build "arm64"
-build "x86_64"
-build "i386"
+  if [[ "${ARCH}" == "armv7" || "${ARCH}" == "armv7s" ]]; then
+     TARGET_CPU="ARM"
+  else
+     TARGET_CPU=`echo -n ${ARCH} | awk '{ print toupper($0) }'`
+  fi
+  
+  if [[ "${PLATFORM}" == "iPhoneSimulator" ]]; then
+    PARAMS="-DTARGET_OS_SIMULATOR=1 -DTARGET_CPU_${TARGET_CPU} -mios-simulator-version-min=8.2  "
+  else
+    PARAMS="-DTARGET_OS_EMBEDDED=1 -DTARGET_CPU_${TARGET_CPU} -miphoneos-version-min=8.2 "
+  fi
+  
+  cd $BUILD_DIR
+  mkdir -p "${PLATFORM}_${ARCH}"
 
-echo $CWD
-cd $CWD
+  export CROSS_TOP="${DEVELOPER}/Platforms/${PLATFORM}.platform/Developer"
+  export CROSS_SDK="${PLATFORM}${SDK_VERSION}.sdk"
 
+  export PARAMS=${PARAMS}"-DSRPC_WITHOUT_IN_QUEUE -DSRPC_WITHOUT_OUT_QUEUE -DSPROTO_WITHOUT_OUT_BUFFER -DUSE_DEPRECATED_EMEV_V1 -DTARGET_OS_IOS=1 -fembed-bitcode -I${IOS_OPENSSL_LIB_INC} -arch ${ARCH} -isysroot ${CROSS_TOP}/SDKs/${CROSS_SDK}"
+  
+  make clean > ${LOG_FILE} 2>&1
+  make all >> ${LOG_FILE} 2>&1
+  RC=$?
+  if [ $RC -ne 0 ]
+  then
+    echo "[ERROR] Make failed with rc=${RC}, please check logs ${LOG_FILE}"
+    exit 1
+  fi
+
+  mv libsupla-client.a "${PLATFORM}_${ARCH}/"
+}
+
```

Comment:
Add:

mkdir -p ${BUILD_DIR}/include
cp ../src/supla-client.h ${BUILD_DIR}/include/
cp ../../supla-common/proto.h ${BUILD_DIR}/include/

Reply:
I added it once after building all versions